### PR TITLE
Kotlinx date library conditional generation

### PR DESCRIPTION
**Pull request by fullkomnun**
prevent kotlinx.date related imports and 'UseSerializers' annotations from being generated for every model even it is unused
and renamed 'datetime.kt' infrastructure util (to a standard name)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 @file:Suppress("UnusedImport", "RemoveRedundantQualifierName")
 
 package {{modelPackage}}
 
-import kotlinx.serialization.UseSerializers
-import {{packageName}}.infrastructure.LocalDateTimeSerializer
-import {{packageName}}.infrastructure.InstantSerializer
-import {{packageName}}.infrastructure.LocalDateSerializer
+{{#dateLibraryKotlinx}}
+    import kotlinx.serialization.UseSerializers
```
Comment (cromefire) [inline modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache]: You have to look out for the indentation, best not use a plugin for mustache. If it's indented here it also will be in the final file.

**Review by cromefire — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache (hunk 1 above)
Comment (fullkomnun) [reply modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache]: Done

File: modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
 {{#imports}}import {{import}}
 {{/imports}}
 
 {{#models}}
-{{#model}}
-{{#isEnum}}{{>enum_class}}{{/isEnum}}{{^isEnum}}{{#isAlias}}typealias {{classname}} = {{{dataType}}}{{/isAlias}}{{^isAlias}}{{>data_class}}{{/isAlias}}{{/isEnum}}
-{{/model}}
+    {{#model}}
+        {{#isEnum}}{{>enum_class}}{{/isEnum}}{{^isEnum}}{{#isAlias}}typealias {{classname}} = {{{dataType}}}{{/isAlias}}{{^isAlias}}{{>data_class}}{{/isAlias}}{{/isEnum}}
+    {{/model}}
```
Comment (cromefire) [inline modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache]: There's the same problem here

**Review by cromefire — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache (hunk 2 above)
Comment (cromefire) [reply modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache]: Forgot to mark all cases

Comment (fullkomnun) [reply modules/openapi-generator/src/main/resources/kotlin-multiplatform-client/model.mustache]: done

**Review by cromefire — APPROVED** [review]

**Pull request merged**